tests/base.py

```diff
@@ -221,7 +221,7 @@ def parse_date(self, date_value):
                 date_stripped = dt.strptime(date_value, "%Y-%m-%dT%H:%M:%SZ")
                 return date_stripped
             except ValueError:
-                raise NotImplementedError
+                return date_value
```

Comment:
Why would we return the date_value here and not just let the error bubble up? In fact, I'm a bit confused on this method for parse_date. It tries to parse in one format, if that fails parses in another? That seems like a bad code smell to me. Are we using it in different places that should be their own method for different fields?

Reply:
I broke this method into 2. One that passes the record back if it wasn't a datetime. The other returns the error. The one that passes records through is used by the modify record method in base. It passes all types in and only parses the ones that fit this datetime formatting. If we are missing a format it will show up as a data discrepancy in the tests, and it can be added later.